third_party/nvidia/lib/Dialect/NVWS/Transforms/Utilities.cpp

```diff
@@ -0,0 +1,50 @@
+#include "Utilities.h"
+#include "triton/Dialect/TritonNvidiaGPU/IR/Dialect.h"
+
+using namespace mlir::triton;
+using namespace mlir::triton::gpu;
+using namespace mlir::triton::nvidia_gpu;
+
+namespace mlir {
+namespace triton {
+namespace nvws {
+
+MemDescType getDataMemDescType(MemDescType memDescType, bool mutableMemory) {
+  auto shape = memDescType.getShape();
+  SmallVector<int64_t> dataShape(shape.begin() + 1, shape.end());
+  return MemDescType::get(dataShape, memDescType.getElementType(),
+                          memDescType.getEncoding(),
+                          memDescType.getMemorySpace(), mutableMemory);
+};
+
+Operation *createAlloc(OpBuilder &builder, Location loc,
+                       MemDescType memDescType, Value src) {
+  if (isa<SharedMemorySpaceAttr>(memDescType.getMemorySpace()))
+    return builder.create<LocalAllocOp>(loc, memDescType, src);
+  else {
+    assert(isa<TensorMemorySpaceAttr>(memDescType.getMemorySpace()));
+    return builder.create<TMEMAllocOp>(loc, memDescType, src);
+  }
+}
+
+MemDescType getArefbufMemDescType(MemDescType memDescType, int32_t depth) {
+  auto shape = memDescType.getShape();
+  SmallVector<int64_t> bufferShape(shape.begin(), shape.end());
+  bufferShape.insert(bufferShape.begin(), depth);
+  return MemDescType::get(bufferShape, memDescType.getElementType(),
+                          memDescType.getEncoding(),
+                          memDescType.getMemorySpace(), true);
+}
```

Comment:
Can you actually move this to `PipeliningUtility.h` and name it something like `getMultibufferedType`?